src/isomorphic/React.js

```diff
@@ -86,4 +71,35 @@ var React = {
   version: ReactVersion,
 };
 
+if (__DEV__) {
+  let warnedForCreateMixin = false;
+  let warnedForCreateClass = false;
+
+  React.createMixin = function(mixin) {
+    warning(
+      warnedForCreateMixin,
+      'React.createMixin is deprecated and should not be used. You ' +
+        'can use this mixin directly instead.',
+    );
+    warnedForCreateMixin = true;
+    return mixin;
+  };
+
+  if (canDefineProperty) {
+    Object.defineProperty(React, 'createClass', {
+      get: function() {
+        warning(
+          warnedForCreateClass,
```

Comment:
We can do that in 15.5 (I have a separate branch that I haven't pushed to yet) but in 16 `React.createClass` is undefined. We could use a void function and warn in there but that would lead to confusing errors in prod.